tensorflow_datasets/scripts/replace_fake_images.py

```diff
@@ -80,14 +83,9 @@ def rewrite_zip(root_dir, zip_filepath):
       zip_file.extractall(path=temp_dir)
 
     rewrite_dir(temp_dir)  # Recursivelly compress the archive content
-
-    # Compressed the .zip file again
-    with zipfile.ZipFile(zip_filepath, 'w') as zip_file:
-      for file_dir, _, files in os.walk(temp_dir):
-        for file in files:
-          file_path = os.path.join(file_dir, file)
-          zip_file.write(file_path)
-
+    
+    shutil.make_archive(zip_filepath[:-4], 'zip', temp_dir) #Copies the directory structure of the extracted file ans stores it back as zip
```

Comment:
I think we should keep using the low level Zip file API as it allow to access more fine grain compression options in the zipfile API.
Is it possible to solve the absolute path bug while keeping the low level API ?

Reply:
1. Sure, I will update the PR with the fake_data dir result!
2. The only benefit of using the Zipfile API which shutil uses as well is that you would have access to the Zip.deflated option.
3. Thank you for pointing out the problem with using [:-4]. My bad ! I will make the correction.
4. I understand that the reason for the fake_data dir is to replicate the original and thus it makes sense to not convert from tar to tar.gz. Will make that change as well.

Reply:
Hey @Conchylicultor, sorry I did not reply to your last comment. Sure, I think it can be done, the only problem here is that if I set the arcname to the filename, directory information would be lost. However, in my opinion, nothing some lengthy 'os' operations can't solve.
The low-level zipfile API can be used, albeit with a loss in readability and not too great of an advantage as far as I can tell. (Maybe you have some inputs regarding this)